src/google/protobuf/compiler/cpp/parse_function_generator.cc

```diff
@@ -1374,7 +1374,7 @@ void ParseFunctionGenerator::GenerateFieldBody(
             "$uint64$ val = ::$proto_ns$::internal::ReadVarint64(&ptr);\n"
```

Comment:
I believe the more correct change is to switch this to `ReadVarint32` instead of `ReadVarint64`.

Reply:
Thanks for the note @fowles! Changing this to `ReadVarint32` would make `val` a `uint32_t`. The call to `*_IsValid` on line 1377 would still require a `static_cast` to an `int`, correct?

Are there any unintended consequences of changing this to `ReadVarint32` (e.g. compatibility)?

Reply:
Yes, you would still need a `static_cast` to int.  The switch to `ReadVarint32` would produce the same behavior as the existing code from a compatibility perspective.

Reply:
Thanks for confirming! I've made the change and updated the PR.